addon-test-support/index.js

```diff
@@ -0,0 +1,57 @@
+import { get } from '@ember/object';
+import { getContext, settled } from '@ember/test-helpers';
+import Test from 'ember-simple-auth/authenticators/test';
+
+const SESSION_SERVICE_KEY = 'service:session';
+const TEST_CONTAINER_KEY = 'authenticator:test';
+
+function ensureAuthenticator(owner) {
+  const authenticator = owner.lookup(TEST_CONTAINER_KEY);
+  if (!authenticator) {
+    owner.register(TEST_CONTAINER_KEY, Test);
+  }
+}
+
+/**
+ * Authenticates the session.
+ *
+ * @param {Object} sessionData Optional argument used to mock an authenticator
+ * response (e.g. a token or user).
+ * @return {Promise}
+ * @public
+ */
+export function authenticateSession(sessionData) {
+  const { owner } = getContext();
+  const session = owner.lookup(SESSION_SERVICE_KEY);
+  owner.setupRouter(); // router must initialize fully before authentication
```

Comment:
I never heard of this method before. Apparently it's private: https://github.com/emberjs/ember.js/blob/74cafbc3593319e803a2b999b99229eb0265235d/packages/ember-application/lib/system/application-instance.js#L165

@rwjblue Do we have a better way of initializing the router? Could that method be public? This is the only dubious thing I can see in the PR.

Reply:
@cibernox `setupRouter()` was added to address comment at https://github.com/simplabs/ember-simple-auth/pull/1519#issuecomment-359837396, i was seeing cases where a route had to be navigated to in order to initialize the router prior to calling `authenticateSession`. This fixed that problem, but would be interested to know from @rwjblue if there is a better way to do it.

Reply:
Yeah, I get why the routes has to be initialized, I'm only wondering if some wise elder can clarify if it's the best approach. 

Also, will this even works in non-application tests?? 

Usually you don't need to authenticate for renderingTests, but I guess it's not crazy to want to use this method to just setup some initial state on the session service. It would be good to at least test that it doesn't explode.